plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -875,17 +905,47 @@ public void revokeRoles(Set<String> roles, Set<HivePrincipal> grantees, boolean
     }
 
     private void invalidatePartitionCache(String databaseName, String tableName)
+    {
+        invalidatePartitionCache(databaseName, tableName, Optional.empty());
+    }
+
+    private void invalidatePartitionCache(String databaseName, String tableName, Optional<Predicate<Optional<String>>> partitionPredicate)
     {
         HiveTableName hiveTableName = hiveTableName(databaseName, tableName);
-        partitionCache.asMap().keySet().stream()
-                .filter(partitionName -> partitionName.getKey().getHiveTableName().equals(hiveTableName))
-                .forEach(partitionCache::invalidate);
-        partitionFilterCache.asMap().keySet().stream()
-                .filter(partitionFilter -> partitionFilter.getKey().getHiveTableName().equals(hiveTableName))
-                .forEach(partitionFilterCache::invalidate);
-        partitionStatisticsCache.asMap().keySet().stream()
-                .filter(partitionFilter -> partitionFilter.getKey().getHiveTableName().equals(hiveTableName))
-                .forEach(partitionStatisticsCache::invalidate);
+
+        Predicate<WithIdentity<HivePartitionName>> hivePartitionPredicate =
+                partitionName -> partitionName.getKey().getHiveTableName().equals(hiveTableName) &&
+                        (partitionPredicate.isPresent() ? partitionPredicate.get().test(partitionName.getKey().getPartitionName()) : true);
+
+        Predicate<WithIdentity<PartitionFilter>> partitionFilerPredicate =
+                partitionFilter -> partitionFilter.getKey().getHiveTableName().equals(hiveTableName) &&
+                        (partitionPredicate.isPresent() ? checkPartitionPredicateOnPartitionFilter(partitionPredicate.get(), partitionFilter.getKey()) : true);
+
+        partitionCache.asMap().keySet().stream().filter(hivePartitionPredicate).forEach(partitionCache::invalidate);
+        partitionFilterCache.asMap().keySet().stream().filter(partitionFilerPredicate).forEach(partitionFilterCache::invalidate);
+        partitionStatisticsCache.asMap().keySet().stream().filter(hivePartitionPredicate).forEach(partitionStatisticsCache::invalidate);
+    }
+
+    private Boolean checkPartitionPredicateOnPartitionFilter(Predicate<Optional<String>> partitionPredicate, PartitionFilter partitionFilter)
+    {
+        return partitionFilter.getPartitionKeysFilter().getDomains()
+                .map(domains -> partitionPredicate.test(Optional.of(convertDomainToPartitionName(domains))))
+                .orElse(false);
+    }
+
+    private String convertDomainToPartitionName(Map<String, Domain> domains)
```

Comment:
@findepi please confirm if this method scope is valid. I had doubts.

Reply:
The method, as implemented now, can fail in various ways.
Why do we need it at all?

Reply:
It's used during filter out of cache entires to be flushed from Partition Filter Cache basing only on schema name, table name and partition name. Key in this cache is PartitionFilter.
So we match HiveTableName using schema and table. And filter domains converted to partition name to match given partition name.

Reply:
Let's invalidate whole `Partition Filter Cache`.

Reply:
Done